libraries/core_libs/consensus/src/dag/block_proposer.cpp

```diff
@@ -162,7 +162,7 @@ SharedTransactions BlockProposer::getShardedTrxs() {
     return {};
   }
 
-  SharedTransactions to_be_packed_trx = trx_mgr_->packTrxs(bp_config_.transaction_limit);
+  SharedTransactions to_be_packed_trx = trx_mgr_->packTrxs();
```

Comment:
not ideal. If you have 10k txs in queue, you will do 10k copies of shared_pointers and put them into newly allocated vector here. Most of them are not needed because you will later use only few hundred of these txs... We could somehow change the code:

- either we will calculate tx gas while getting it from tx queue (this would block the queue for long time as there is a lock)
- get txs in batches by (250/500 ?), calculate the estimation and if didnt reach the limit then get another batch... 
- by knowing the tx with smallest gas, we could know how many such txs could be in single block and we can ask that many txs from queue (not all of them like now)

Reply:
I refactored this part, please review again